src/sagemaker/image_uris.py

```diff
@@ -387,9 +424,12 @@ def _validate_arg(arg, available_options, arg_name):
         )
 
 
-def _format_tag(tag_prefix, processor, py_version, container_version):
+def _format_tag(tag_prefix, processor, py_version, container_version, inference_tool=None):
     """Creates a tag for the image URI."""
-    return "-".join(x for x in (tag_prefix, processor, py_version, container_version) if x)
+    if not inference_tool:
+        return "-".join(x for x in (tag_prefix, processor, py_version, container_version) if x)
+
+    return "-".join(x for x in (tag_prefix, inference_tool, py_version, container_version) if x)
```

Comment:
nit: you can remove not here and swap returns

Reply:
updated.